src/vm/methodtablebuilder.cpp

```diff
@@ -4222,6 +4222,12 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
                 // Inherit IsByRefLike characteristic from fields
                 if (!IsSelfRef(pByValueClass) && pByValueClass->IsByRefLike())
                 {
+                    if (fIsStatic || !IsValueClass())
+                    {
+                        // fIsStatic - By-ref-like types cannot be used for static fields
+                        // !IsValueClass() - Non-value-classes cannot contain by-ref-like instance fields
+                        COMPlusThrowHR(COR_E_TYPELOAD);
```

Comment:
It may be useful to add more descriptive error message for this - I expect that folks are going to hit this until we get the enforcement in the language.

```
BuildMethodTableThrowException(IDS_XXX);
```